clients/web/src/views/widgets/MarkdownWidget.js

```diff
@@ -13,19 +146,35 @@ girder.views.MarkdownWidget = girder.View.extend({
      * @param [settings.placeholder=''] Text area placeholder.
      * @param [settings.prefix='markdown'] Prefix for element IDs in case
      *     multiple of these widgets are rendered simultaneously.
+     * @param [settings.enableUploads=false] Whether to allow uploading files
+     *     inline into the markdown.
+     * @param [settings.maxUploadSize=null] Max upload size in bytes.
+     * @param [settings.parent] If enableUploads is true, set this to the object
+     *     into which files should be uploaded.
+     * @param [settings.allowedExtensions=null] If you want to restrict to
+     *     certain file extensions, set this to a list of (lowercase) extensions,
+     *     not including periods. E.g.: ["png", "jpg", "jpeg"]
      */
     initialize: function (settings) {
         this.text = settings.text || '';
         this.placeholder = settings.placeholder || '';
         this.prefix = settings.prefix || 'markdown';
+        this.enableUploads = settings.enableUploads || false;
+        this.parent = settings.parent || null;
+        this.maxUploadSize = settings.maxUploadSize || null;
```

Comment:
I'm not sure the default of this should be `null`.  Though I suppose only people with access to upload to the folder can edit this anyway, so it doesn't add any protection really.

Amusingly, I tried to upload a 4Gb file (renamed to .jpg).  This uploaded and now is downloading for the preview, at which point my Chrome tab died.  I'd recommend a default of 10 Mbytes (anything larger used to kill iOS 5, perhaps it is higher today).